### PR TITLE
ROU-3349: InlineSVG - Implement with the new architecture

### PR DESCRIPTION
**Pull request by OS-giulianasilva**
This PR is for implement the new InlineSVG

### What was done
- Created the OSFramework.OSUI.Patterns.InlineSvg.InlineSvg
   - Now the Initialized event should be triggered after component's build.
   - Now the SVGCode in validated before assign it to the config.
- Created the OSFramework.OSUI.Patterns.InlineSvg.InlineSvgConfig
- Created the OSFramework.OSUI.Helper.SVG that contains the isValid method to validate the SVGCode.
- Created the OutSystems.OSUI.Patterns.InlineSvgAPI
- Added the necessary interfaces and errorCodes.

### Checklist

-   [x] tested locally
-   [x] documented the code
-   [x] clean all warnings and errors of eslint
-   [x] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/IInlineSvg.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace OSFramework.OSUI.Patterns.InlineSvg {
+	/**
+	 * Defines the interface for OutSystemsUI InlineSvg Pattern
+	 */
+	export interface IInlineSvg extends Interface.IPattern {
+		registerCallback(eventName: string, callback: GlobalCallbacks.OSGeneric): void;
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/InlineSvg/IInlineSvg.ts:7]: You don't need this method here,  you can add the the Interface.ICallback, after the IPattern

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		/**
+		 * Set the html references that will be used to manage the cssClasses and atribute properties
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.InlineSvg.InlineSvg
+		 */
+		protected setHtmlElements(): void {
+			this._parentSelf = Helper.Dom.GetElementById(this.widgetId);
+
+			Helper.AsyncInvocation(this._platformEventOnInitialize, this.widgetId);
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts:54]: I would trigger this after the this.finishBuild(), on the build() method

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+		/**
+		 * Reassign the HTML elements to undefined, preventing memory leaks
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.InlineSvg.InlineSvg
+		 */
+		protected unsetHtmlElements(): void {
+			this._parentSelf = undefined;
+			this._platformEventOnInitialize = undefined;
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts:75]: The this._platformEventOnInitialize = undefined should be on the unsetCallbacks() method.

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+		 * Destroy the InlineSvg
+		 *
+		 * @memberof OSFramework.Patterns.InlineSvg.InlineSvg
+		 */
+		public dispose(): void {
+			if (this.isBuilt) {
+				// Remove unused HTML elements
+				this.unsetHtmlElements();
+
+				//Destroying the base of pattern
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts:126]: missing unsetCallbacks() call

**Review by BenOsodrac — CHANGES_REQUESTED** [review]

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/IInlineSvg.ts (hunk 1 above)
Comment (OS-giulianasilva) [reply src/scripts/OSFramework/OSUI/Pattern/InlineSvg/IInlineSvg.ts]: Fixed

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts (hunk 2 above)
Comment (OS-giulianasilva) [reply src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts]: Fixed

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts (hunk 3 above)
Comment (OS-giulianasilva) [reply src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts]: Fixed

File: src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts (hunk 4 above)
Comment (OS-giulianasilva) [reply src/scripts/OSFramework/OSUI/Pattern/InlineSvg/InlineSvg.ts:127]: Fixed

**Review by BenOsodrac — APPROVED** [review]

File: src/scripts/OSFramework/OSUI/Helper/SVG.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		/**
+		 * Function that validates if a given svgString is a valid SVG
+		 *
+		 * @param url
+		 * @memberof OSFramework.Helper.URL
+		 */
+		public static IsValid(svgString: string): boolean {
+			const parser = new DOMParser();
+			try {
+				const doc = parser.parseFromString(svgString, 'image/svg+xml');
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Helper/SVG.ts:14]: In order to keep consistency, I suggest to Enum the text on method.
- 'image/svg+xml'
- 'parsererror'
- 'svg'

Comment (OS-giulianasilva) [reply src/scripts/OSFramework/OSUI/Helper/SVG.ts]: done

**Review by bmarcelino-fe — APPROVED** [review]

**Review by BenOsodrac — APPROVED** [review]

**Pull request merged**